unittests/ThunkLibs/common.h

```diff
@@ -19,8 +19,14 @@ class TestDiagnosticConsumer : public clang::TextDiagnosticPrinter {
 
 public:
   TestDiagnosticConsumer(bool silent_)
-    : clang::TextDiagnosticPrinter(llvm::errs(), new clang::DiagnosticOptions)
-    , silent(silent_) {}
+    :
+#if LLVM_VERSION_MAJOR >= 21
+    clang::TextDiagnosticPrinter(llvm::errs(), *(new clang::DiagnosticOptions))
```

Comment:
I must've misunderstood the ownership semantics when writing this code; storing the DiagnosticOptions as a `unique_ptr` member so it doesn't leak after each test would be preferable.

Reply:
Made a drive-by fix